dm/pkg/checker/privilege.go

```diff
@@ -165,6 +165,49 @@ func (pc *SourceReplicatePrivilegeChecker) Name() string {
 	return "source db replication privilege checker"
 }
 
+type TargetPrivilegeChecker struct {
+	db     *sql.DB
+	dbinfo *dbutil.DBConfig
+}
+
+func NewTargetPrivilegeChecker(db *sql.DB, dbinfo *dbutil.DBConfig) RealChecker {
+	return &TargetPrivilegeChecker{db: db, dbinfo: dbinfo}
+}
+
+func (t *TargetPrivilegeChecker) Name() string {
+	return "target db privilege checker"
+}
+
+func (t *TargetPrivilegeChecker) Check(ctx context.Context) *Result {
+	result := &Result{
+		Name:  t.Name(),
+		Desc:  "check privileges of target DB",
+		State: StateSuccess,
+		Extra: fmt.Sprintf("address of db instance - %s:%d", t.dbinfo.Host, t.dbinfo.Port),
+	}
+	grants, err := dbutil.ShowGrants(ctx, t.db, "", "")
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+	replRequiredPrivs := map[mysql.PrivilegeType]priv{
+		mysql.CreatePriv: {needGlobal: true},
+		mysql.SelectPriv: {needGlobal: true},
+		mysql.InsertPriv: {needGlobal: true},
+		mysql.UpdatePriv: {needGlobal: true},
+		mysql.DeletePriv: {needGlobal: true},
+		mysql.AlterPriv:  {needGlobal: true},
+		mysql.DropPriv:   {needGlobal: true},
+		mysql.IndexPriv:  {needGlobal: true},
+	}
```

Comment:
I'm not sure if we need a Super privilege to some precheck stuff (e.g. `SHOW PROCESSLIST` needs super privilege to show other users' connections🤔)